libraries/AP_Math/AP_Math.cpp

```diff
@@ -339,14 +339,13 @@ uint16_t get_random16(void)
 }
 
 
-#if AP_SIM_ENABLED
-// generate a random float between -1 and 1, for use in SITL
+// generate a random float between -1 and 1
 float rand_float(void)
 {
 #if CONFIG_HAL_BOARD == HAL_BOARD_SITL
     return ((((unsigned)random()) % 2000000) - 1.0e6) / 1.0e6;
 #else
-    return get_random16() / 65535.0;
+    return ((get_random16() / 65535.0) -1) * 2;
```

Comment:
```suggestion
    return ((get_random16() / 65535.0) - 0.5) * 2;
```

You fixed it too well, now it can’t return positive numbers.

Reply:
Man.  Order of operations failure.

I've patched this.  And added a test....